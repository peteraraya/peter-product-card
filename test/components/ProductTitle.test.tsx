import * as React from 'react';
import renderer from 'react-test-renderer';
import { ProductTitle, ProductCard } from '../../src/components';
import { product1 } from '../data/products';

describe('ProductTitle', () => {
  test('debe mostrar el componente correctamente con el titulo personalizado', () => {
    const wrapper = renderer.create(
      <ProductTitle title="Titulo personalizado" className="custom-class" />
    );

    expect(wrapper).toMatchSnapshot();
    // console.log(wrapper.toJSON());
  });

  test('debe mostrar el componente correctamente con el nombre del producto', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1} initialValues={{ count: 6 }}>
        {() => <ProductTitle />}
      </ProductCard>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
