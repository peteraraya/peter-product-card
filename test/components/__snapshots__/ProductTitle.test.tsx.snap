// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle debe mostrar el componente correctamente con el nombre del producto 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - sin imagen
  </span>
</div>
`;

exports[`ProductTitle debe mostrar el componente correctamente con el titulo personalizado 1`] = `
<span
  className="productDescription custom-class"
>
  Titulo personalizado
</span>
`;
